src/Altinn.App.Core/Infrastructure/Clients/Authorization/AuthorizationClient.cs

```diff
@@ -179,4 +180,52 @@ List<string> actions
         }
         return MultiDecisionHelper.ValidatePdpMultiDecision(actionsResult, response.Response, user);
     }
+
+    /// <summary>
+    /// Retrieves roles for a user on a specified party.
+    /// </summary>
+    /// <param name="userId">The user id.</param>
+    /// <param name="userPartyId">The user party id.</param>
+    /// <returns>A list of roles for the user on the specified party.</returns>
+public async Task<List<Role>> GetUserRolesAsync(int userId, int userPartyId)
+{
+    List<Role> roles = new();
```

Comment:
Usually we wrap the body in activity (distributed trace spans). You can probably find example usages of `Telemetry`